modules/javafx.graphics/src/main/java/com/sun/javafx/text/PrismTextLayout.java

```diff
@@ -421,90 +421,41 @@ public PathElement[] getCaretShape(int offset, boolean isLeading,
 
     @Override
-    public Hit getHitInfo(float x, float y, String text, int textRunStart, int curRunStart) {
+    public Hit getHitInfo(float x, float y) {
         int charIndex = -1;
         int insertionIndex = -1;
         boolean leading = false;
-        int relIndex = 0;
-        int textWidthPrevLine = 0;
 
         ensureLayout();
-        int lineIndex = getLineIndex(y, text, curRunStart);
+        int lineIndex = getLineIndex(y);
         if (lineIndex >= getLineCount()) {
             charIndex = getCharCount();
             insertionIndex = charIndex + 1;
         } else {
-            if (isMirrored()) {
-                x = getMirroringWidth() - x;
-            }
             TextLine line = lines[lineIndex];
             TextRun[] runs = line.getRuns();
             RectBounds bounds = line.getBounds();
             TextRun run = null;
             x -= bounds.getMinX();
             //TODO binary search
-            if (text == null || spans == null) {
-                for (int i = 0; i < runs.length; i++) {
-                    run = runs[i];
-                    if (x < run.getWidth()) break;
-                    if (i + 1 < runs.length) {
-                        if (runs[i + 1].isLinebreak()) break;
-                        x -= run.getWidth();
-                    }
-                }
-            } else {
-                for (int i = 0; i < lineIndex; i++) {
-                    for (TextRun r: lines[i].runs) {
-                        if (r.getTextSpan() != null && r.getStart() >= textRunStart && r.getTextSpan().getText().equals(text)) {
-                            textWidthPrevLine += r.getLength();
-                        }
-                    }
-                }
-                int prevNodeLength = 0;
-                boolean isPrevNodeExcluded = false;
-                for (TextRun r: runs) {
-                    if (!r.getTextSpan().getText().equals(text) || (r.getStart() < textRunStart && r.getTextSpan().getText().equals(text))) {
-                        prevNodeLength += r.getWidth();
-                        continue;
-                    }
-                    if (r.getTextSpan() != null && r.getTextSpan().getText().equals(text)) {
-                        BaseBounds textBounds = new BoxBounds();
-                        getBounds(r.getTextSpan(), textBounds);
-                        if (textBounds.getMinX() == 0 && !isPrevNodeExcluded) {
-                            x -= prevNodeLength;
-                            isPrevNodeExcluded = true;
-                        }
-                        if (x > r.getWidth()) {
-                            x -= r.getWidth();
-                            relIndex += r.getLength();
-                            continue;
-                        }
-                        run = r;
-                        break;
-                    }
+            for (int i = 0; i < runs.length; i++) {
+                run = runs[i];
+                if (x < run.getWidth()) break;
```

Comment:
I would strongly recommend placing `break`s on separate lines for convenience during debugging.